src/interfaces/IButteredBread.sol

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.25;
+
+/**
+ * @title Breadchain Buttered Bread interface
+ */
+interface IButteredBread {
+    /// @notice Occurs when user does not have sufficient Butter to mint ButteredBread
+    error InsufficientFunds();
+    /// @notice Occurs when an invalid value is attempted to be used in setter functions
+    error InvalidValue();
+    /// @notice Occurs when attempting a deposit with a non sanctioned LP
+    error NotAllowListed();
+    /// @notice Occurs when attempting to transfer ButteredBread , a utility token for voting not meant for trading
+    error NonTransferable();
+
+    /// @notice Specifics how much LP Token (Butter) has been added
+    event AddButter(address _account, address _lp, uint256 _amount);
+    /// @notice Specifics how much LP Token (Butter) has been removed
+    event RemoveButter(address _account, address _lp, uint256 _amount);
+
+    /**
+     * @param liquidityPools sanctioned LPs
+     * @param scalingFactors scaling factor on mint per sanctioned LP
+     * Note: each scaling factor is a fixed point percent (e.g. 100 = 1X, 150 = 1.5X, 1000 = 10X)
+     * @param name ERC20 token name
+     * @param symbol ERC20 token symbol
+     */
+    struct InitData {
+        address[] liquidityPools;
+        uint256[] scalingFactors;
+        string name;
+        string symbol;
+    }
+
+    /**
+     * @param balance Value of deposited Butter (LP tokens)
+     * @param scalingFactor At the time of deposit or updated with `syncVotingWeight` function
+     */
+    struct LPData {
+        uint256 balance;
+        uint256 scalingFactor;
+    }
+
+    /// @notice initialize contract as TransparentUpgradeableProxy
+    function initialize(InitData calldata _initData) external;
+
+    /// @notice Returns whether a given liquidity pool is Breadchain sanctioned or not
+    function allowlistedLPs(address _lp) external view returns (bool _allowed);
+
+    /// @notice Returns the factor that determines how much ButteredBread should be minted for a Liquidity Pool token (Butter)
+    function scalingFactors(address _lp) external view returns (uint256 _factor);
+
+    /// @notice The amount of LP tokens (Butter) deposited for a an account
+    function accountToLPData(address _account, address _lp) external view returns (LPData memory _lpData);
```

Comment:
Would rather see this broken out into two functions that return the two pieces of data explicitly like:

"balance(account, lp)"
"scalingFactor(account, lp)"

It allows us to remove the interfaces need to have context on "LPData" struct so that it can be moved into the contract definition, which imo makes the contract more readable and easier to work with. `accountToLPData` can just be made public if we want to make it accessible like this.

Reply:
we need the LPData struct for math. you can use the `accountToLPBalance` function to view Butter balance

Reply:
also, `public` does not work well with custom struct mappings, hence why i made it internal and added a custom view function